### DISCUSSION TITLE
Cache config updated in Keycloak.X version not getting applied

### DISCUSSION THREAD

**Discussion by sreekesh93** [keycloak/keycloak#11503] [Archive]
To increase realm & user cache size , respective values are updated in cache-ispn.xml file and copied at time of building docker image
[dockerfile.txt](https://github.com/keycloak/keycloak/files/8514029/dockerfile.txt)

on running this we could notice the cache.xml file is there in conf folder. But this values are not getting applied. Am I missing some thing? how to check the updated cache size is got applied  or not in Keycloak.x version (17.0.1)

Other env values:
  - name: KC_PROXY
    value: edge 
  - name: KC_CACHE_STACK
    value: kubernetes
  - name: KC_CACHE
    value: ispn

**Comment by sreekesh93** [comment]
I tried to enable jmx to check on cache statistics, I tried to do this in local setup with following command
docker run -it --rm --name keycloak \
  --cap-add SYS_ADMIN \
  -p 8080:8080 \
  -p 8787:8787 \
  -p 8999:8999 \
  -e KEYCLOAK_ADMIN="keycloak" \
  -e KEYCLOAK_ADMIN_PASSWORD="keycloak" \
  -e DEBUG="true" \
  -e DEBUG_PORT="*:8787" \
  -e JAVA_OPTS_APPEND="-Xmx1g \
    -Dcom.sun.management.jmxremote.port=8999 -Dcom.sun.management.jmxremote.rmi.port=8999 \
    -Dcom.sun.management.jmxremote.ssl=false -Dcom.sun.management.jmxremote.authenticate=false \
    -Dcom.sun.management.jmxremote.local.only=false -Djava.rmi.server.hostname="$(hostname)" \
    -XX:+HeapDumpOnOutOfMemoryError -XX:HeapDumpPath=/tmp/keycloak.hprof" \
  quay.io/keycloak/keycloak:17.0.0 start-dev \
    --log-level=INFO

Jmx is enabled but not getting connected from jconsole

**Comment by sreekesh93** [comment]
Able to connect with jmx port, the cache got applied

**Reply by davoustp** [reply]
Hi @sreekesh93 ,
Just saw your comment on #11074 , looks like you finally did manage to connect your `jconsole` to the JMX port, correct?

**Reply by sreekesh93** [reply]
Hi @davoustp , Yes able to connect with JMX port, Able to notice that realm cache size got update from default 10K to 100K. Thanks for your reply.

By updating the realm cache size to 100K for v16 & v17 legacy we were able to reach 1500realm creation, in which the realm creation time gradually increased. On applying same config the quarkus is able to create only 1100 realms. Do we need to tweak any quarkus configs like core-thread-pool or anything else?

**Reply by davoustp** [reply]
> By updating the realm cache size to 100K for v16 & v17 legacy we were able to reach 1500realm creation, in which the realm creation time gradually increased.

Yes, that's what #11074 is all about.

> On applying same config the quarkus is able to create only 1100 realms.
I did not test Wildfly-based Keycloak (legacy), only the Quarkus-based one.

But you are very patient to reach 1100 realms - I stopped at ~600 realms (but did not increase the cache size while doing so, that's true). :-)

> Do we need to tweak any quarkus configs like core-thread-pool or anything else?

I did not change anything else on Quarkus config in my investigation.
But I did change the code, so I don't think it can compare to stock Keycloak...

**Reply by sreekesh93** [reply]
Thanks for your reply @davoustp

**Reply by davidfrickert** [reply]
can you share you infinispan xml file @sreekesh93 ?
i'm trying to access infinispan MBeans in KC 18 with no sucess.
I have enabled statistics in the `cache-container` and in the caches themselves but it doesn't seem to work. 
```
<cache-container name="keycloak" statistics="true">
    <local-cache name="realms" statistics="true">
(...)
```

**Reply by davidfrickert** [reply]
> can you share you infinispan xml file @sreekesh93 ? i'm trying to access infinispan MBeans in KC 18 with no sucess. I have enabled statistics in the `cache-container` and in the caches themselves but it doesn't seem to work.
> 
> ```
> <cache-container name="keycloak" statistics="true">
>     <local-cache name="realms" statistics="true">
> (...)
> ```

it seems i was missing something like this:
```
<cache-container name="keycloak" statistics="true">
    <jmx enabled="true" domain="infinispan"/>
```
